overlord/devicestate/handlers_gadget.go

```diff
@@ -210,11 +212,78 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 	return snapstate.FinishTaskWithRestart(t, state.DoneStatus, restart.RestartSystem, nil)
 }
 
-func (m *DeviceManager) updateGadgetCommandLine(t *state.Task, st *state.State, isUndo bool) (updated bool, err error) {
-	snapsup, err := snapstate.TaskSnapSetup(t)
+// fromSystemOption tells us if t was created when setting a system
+// option for the kernel command line.
+func fromSystemOption(t *state.Task) (bool, error) {
+	for _, param := range []string{"system.kernel.cmdline-append", "system.kernel.dangerous-cmdline-append"} {
```

Comment:
we changed these names since, we probably need unit tests here? and maybe even an integration test in managers_test?